src/classic/element/ReactElementValidator.js

```diff
@@ -213,19 +199,21 @@ function validateChildKeys(node, parentType) {
   } else if (node) {
     var iteratorFn = getIteratorFn(node);
     // Entry iterators provide implicit keys.
-    if (iteratorFn && iteratorFn !== node.entries) {
-      var iterator = iteratorFn.call(node);
-      var step;
-      while (!(step = iterator.next()).done) {
-        if (ReactElement.isValidElement(step.value)) {
-          validateExplicitKey(step.value, parentType);
+    if (iteratorFn) {
```

Comment:
Note that this may break some use cases people have when using Map / OrderedMap with v13-beta/master. Nothing wrong with that, but something people need to be aware of.

Since Maps are iterable, and they yield `[key, value]` tuples, you'll start to see your Map keys showing up in the UI.

No clear fix with ES6 Map, unfortunately (not that anyone is using them in this way), but for Immutable.js Map you'll need to call `.toIndexedSeq().map()` instead of just `.map()` with the current behavior.